org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/event/EventRecordingAssert.java

```diff
@@ -0,0 +1,106 @@
+
+	@SuppressWarnings("unchecked")
+	private static <T> Predicate<Object> hasTypeAnd(Class<T> clazz, Predicate<T> predicate) {
+		return e -> clazz.isAssignableFrom(e.getClass()) && predicate.test((T) e);
+	}
+
+	public EventRecordingAssert(EventRecording actual, Class<EventRecordingAssert> selfType) {
+		super(actual, selfType);
+	}
+
+	public static EventRecordingAssert assertThat(EventRecording actual) {
+		return new EventRecordingAssert(actual, EventRecordingAssert.class);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 * @param predicate - Predicate that would be tested against the
+	 *            ServiceEvents
+	 * @param timeout - the time in that the matches must happened
+	 */
+	public static EventRecordingAssert assertThatServiceEvent(Runnable execute, Predicate<ServiceEvent> predicate,
+		int timeout) throws Exception {
+		return assertThat(execute, hasTypeAnd(ServiceEvent.class, predicate), timeout);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 * @param predicate - Predicate that would be tested against the
+	 *            BundleEvents
+	 * @param timeout - the time in that the matches must happened
+	 */
+	public static EventRecordingAssert assertThatBundleEvent(Runnable execute, Predicate<BundleEvent> predicate,
+		int timeout) throws Exception {
+		return assertThat(execute, hasTypeAnd(BundleEvent.class, predicate), timeout);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 * @param predicate - Predicate that would be tested against the
+	 *            FrameworkEvents
+	 * @param timeout - the time in that the matches must happened
+	 */
+	public static EventRecordingAssert assertThatFrameworkEvent(Runnable execute, Predicate<FrameworkEvent> predicate,
+		int timeout) throws Exception {
+		return assertThat(execute, hasTypeAnd(FrameworkEvent.class, predicate), timeout);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 * @param predicate - Predicate that would be tested against the Events
+	 * @param timeout - the time in that the matches must happened
+	 */
+	public static EventRecordingAssert assertThat(Runnable execute, Predicate<Object> predicate, int timeout)
+		throws Exception {
+		BundleContext bc = FrameworkUtil.getBundle(EventRecordingAssert.class)
+			.getBundleContext();
+
+		EventRecording awaitResult = EventRecording.record(bc, execute, predicate, timeout);
+		return new EventRecordingAssert(awaitResult, EventRecordingAssert.class);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 * @param timeout - the time in that the matches must happened
+	 */
+	public static EventRecordingAssert assertThat(Runnable execute, int timeout) throws Exception {
+		return assertThat(execute, (se) -> false, timeout);
+	}
+
+	/**
+	 * @param execute - the action that will be invoked.
+	 */
+	public static EventRecordingAssert assertThat(Runnable execute) throws Exception {
+		return assertThat(execute, 200);
```

Comment:
Where does `200` come from? It seems wrong to hard code a timeout value in an API. Perhaps we remove this method?